### PR TITLE
Sort award achievement list alphabetically by student name.

### LINKED ISSUE #1582: Award Achievement list is not sorted
Student names are not sorted when viewing the list of students who have obtained an achievement.

### PR DESCRIPTION
**Pull request by fonglh**
Fixes #1582.

### CODE REVIEW

**Comment by weiqingtoh** [comment]
LGTM - random question - do we wanna split up phantom and non-phantom students?

**Comment by fonglh** [comment]
I think no need? Doesn't really make sense to award achievements to phantom students, since they're phantom anyway.

**Comment by weiqingtoh** [comment]
If that's the case, we might need to apply another scope `without_phantom_students`

**Comment by allenwq** [comment]
Do we use exclude or split phantom users in other places ? It should be consisent with them.

**Comment by fonglh** [comment]
Looks like the statistics view splits phantom and non-phantom.
https://github.com/Coursemology/coursemology2/blob/ad4e0068abe23b900d2ab18da3bbf29ee8655098/app/views/course/statistics/student.html.slim

**Comment by allenwq** [comment]
The statistics page is not a form and it's different.

It's not very starght forward to split in this case, simple_form is invovled.  And if we want to split it should be global, all the collection input will need to do that.

**Comment by fonglh** [comment]
@allenwq take a look? `collection` is initialised outside `simple_form` and passed in. Some of the CSS selectors and button text is also passed in from the index page.

**Comment by allenwq** [comment]
OK. I thought it's same as https://github.com/Coursemology/coursemology2/blob/master/app/views/course/groups/_group_user_fields.html.slim#L2 :) In those cases, we didn't split phantom and normal users.

This LGTM.

File: app/views/course/achievement/course_users/index.html.slim [hunk 1]
```diff
@@ -3,4 +3,16 @@
...
+- collection = @course.course_users.with_approved_state.without_phantom_users.students.order_alphabetically
+= render partial: 'course_users_form', locals: { course: @course, achievement: @achievement, collection: collection,
+                                                 css: 'normal', button: 'button' }
+
+br
+h4 = t('.phantom_user_header')
+
+- collection = @course.course_users.with_approved_state.phantom.students.order_alphabetically
+= render partial: 'course_users_form', locals: { course: @course, achievement: @achievement, collection: collection,
+                                                 css: 'phantom', button: 'phantom_button' }
```
Comment (weiqingtoh) [inline app/views/course/achievement/course_users/index.html.slim]: LGTM, but I lean towards less locals into partials in general (to make it is easier for us to understand one) - don't want to have to manage too many variables.

**Pull request merged**